### PR TITLE
feat: use OZ SafeERC20 library

### PR DESCRIPTION
**Pull request by 0xAustrian**
Closes DTT-308

### CODE REVIEW

File: contracts/core/libraries/Transfer.sol [hunk 1]
```diff
@@ -65,9 +65,9 @@ library Transfer {
     function transferAmountFrom(address _token, address _from, address _to, uint256 _amount) internal {
         if (_token == NATIVE) {
             // '_from' is ignored. The contract's balance is used.
-            if (_to != address(this)) _to.safeTransferETH(_amount);
+            if (_to != address(this)) payable(_to).transfer(_amount);
```
Comment (0xOneTony) [inline contracts/core/libraries/Transfer.sol:68]: we should still use safe transfer for native transfers

Comment (0xAustrian) [reply contracts/core/libraries/Transfer.sol:68]: There is no `safeTransferETH` function in OZ library, you mean using both libraries at the same time?

File: contracts/core/libraries/Transfer.sol [hunk 2]
```diff
@@ -2,7 +2,7 @@
 pragma solidity ^0.8.19;
 
 // External Imports
-import {SafeTransferLib} from "solady/utils/SafeTransferLib.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
```
Comment (thelostone-mc) [inline contracts/core/libraries/Transfer.sol:5]: curious . why are we switching ?

Comment (0xOneTony) [reply contracts/core/libraries/Transfer.sol:5]: solady's SafeTransferLib doesnt check for tokens code.length to check if the token is a contract. Also the library does not handle incorrect or unexpected return data from token contracts and some transactions may pass silently, even if the underlying transfer has failed or is not executed correctly.

**Pull request closed without merging**